enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/discovery/HazelcastClusterTopology.java

```diff
@@ -222,7 +222,7 @@ static void casLeaders( HazelcastInstance hazelcastInstance, LeaderInfo leaderIn
 
         boolean greaterOrEqualTermExists = Optional.ofNullable( expected ).map(l -> l.term() >= leaderInfo.term() ).orElse( false );
 
-        if ( greaterOrEqualTermExists || noUpdate )
+        if ( (greaterOrEqualTermExists || noUpdate) && !leaderInfo.isStepDown() )
```

Comment:
Maybe extract `(greaterOrEqualTermExists || noUpdate)` to an appropriately named variable?